rs/tests/dre/utils/steps/update_subnet_type.rs

```diff
@@ -146,6 +148,85 @@ impl Step for UpdateSubnetType {
     }
 }
 
+#[derive(Clone)]
+pub struct UpdateApiBoundaryNodes {
+    pub version: String,
+}
+
+impl Step for UpdateApiBoundaryNodes {
+    fn execute(
+        &self,
+        env: ic_system_test_driver::driver::test_env::TestEnv,
+        rt: Handle,
+    ) -> anyhow::Result<()> {
+        let logger = env.logger();
+
+        // check that there are even API boundary nodes
+        if env
+            .topology_snapshot()
+            .api_boundary_nodes()
+            .next()
+            .is_none()
+        {
+            info!(logger, "There are no API boundary nodes to be upgraded");
+            return Ok(());
+        }
+
+        // all API BNs should be on the same version
+        let versions = get_api_boundary_nodes_version(env.topology_snapshot());
+        let unique_versions: HashSet<_> = versions.values().cloned().collect();
+
+        if unique_versions.len() > 1 {
+            return Err(anyhow::anyhow!(
+                "The API BNs are on different versions: {}",
+                versions
+                    .iter()
+                    .map(|(node_id, version)| format!("{} -> {}", node_id, version))
+                    .join(", ")
+            ));
+        }
+
+        // check whether the current version of the API BNs is already the one they should be upgraded to
+        let current_version = unique_versions.iter().next().unwrap();
+        if current_version.eq(&self.version) {
+            info!(
+                logger,
+                "API boundary nodes are already on version {}", self.version
+            );
+            return Ok(());
+        }
+
+        // try to upgrade them
+        let nns_node = env.get_first_healthy_nns_node_snapshot();
+        let nns = runtime_from_url(nns_node.get_public_url(), nns_node.effective_canister_id());
+        let governance_canister = get_governance_canister(&nns);
+        let test_neuron_id = NeuronId(TEST_NEURON_1_ID);
+        let proposal_sender = Sender::from_keypair(&TEST_NEURON_1_OWNER_KEYPAIR);
+
+        let node_ids: Vec<_> = env
+            .topology_snapshot()
+            .api_boundary_nodes()
+            .map(|n| n.node_id)
+            .collect();
+
+        let proposal_id = rt.block_on(submit_update_api_boundary_node_version_proposal(
```

Comment:
I would also add some logs per `rt.block_on`. That will help later when debugging in case this test times out in one of these operations.